src/gui/properties/trackerlistwidget.cpp

```diff
@@ -466,8 +466,7 @@ void TrackerListWidget::editSelectedTracker()
 
     QList<BitTorrent::TrackerEntry> trackers = torrent->trackers();
     bool match = false;
-    for (int i = 0; i < trackers.size(); ++i) {
-        BitTorrent::TrackerEntry &entry = trackers[i];
+    for (auto &entry : trackers) {
```

Comment:
beware when using similar loops, container can be detached!
from [Qt documentation](https://doc.qt.io/qt-5/containers.html#the-foreach-keyword):
> An alternative to Qt's foreach loop is the range-based for that is part of C++ 11 and newer. However, keep in mind that the range-based for might force a Qt container to detach, whereas foreach would not. But using foreach always copies the container, which is usually not cheap for STL containers. If in doubt, prefer foreach for Qt containers, and range based for for STL ones.

but from [`foreach` documentation](https://doc.qt.io/qt-5/qtglobal.html#foreach):
> Note: Since Qt 5.7, the use of this macro is discouraged. It will be removed in a future version of Qt. Please use C++11 range-for, possibly with qAsConst(), as needed.

so, as result [`qAsConst()`](https://doc.qt.io/qt-5/qtglobal.html#qAsConst) must be used in similar cases to prevent container detaching. see [its documentation](https://doc.qt.io/qt-5/qtglobal.html#qAsConst) for examples, they have a lot of useful comments

Reply:
@Kolcha, how your comment is applicaple to this particular place?

Reply:
sorry, I didn't see line `entry = newEntry;` below... so, there no any issues in this case. I'll delete my my comment later